src/store.ts

```diff
@@ -49,18 +50,31 @@ function bindInitialState(pages: Array<PageLoader<any, any>>): any {
   return initialState
 }
 
-export function createPoetStore() {
-  const pages = pageCreators.map(Page => new Page())
+export function createPoetStore(): Promise<{
+  readonly store: any
+  readonly pages: any
+}> {
+  return new Promise((resolve, reject) => {
```

Comment:
Not sure but I think you can avoid wrapping the entire function in a `new Promise` and `try/catch` if you just declare it as async. 

All async functions always return a promise, and any error thrown inside an async function gets converted into a rejection automatically.

Can you give it a try?

Reply:
This will come in once the CreatePoetStore PR is merged.